### DISCUSSION TITLE
API to allow creating a geofence

### DISCUSSION THREAD

**Discussion by SteveDonie** [tobiasehlert/teslamateapi#291] [Ideas]
Big picture: I have Teslamate and NodeRed set up, I have customed the NodeRed script quite a bit to send me information through Telegram. When I park, the script uses OpenStreetMap to look up the address at my lat/long and put that in a message. If I park somewhere that is 'undefined' but I want to 'remember' that place, I'd like to just type "fence" into Telegram and have it start a mini-conversation:

**Teslamate** parked at undefined North Mopac Service Road, Austin Texas
**me** fence
**Teslamate** What is the name of the new geofence?
**me** Bob's Burgers
**Teslamate** OK, created geofence "Bob's Burgers"

Sound interesting?

**Comment by tobiasehlert** [comment]
Hi @SteveDonie,

Thanks for your suggestion!
I think spontaneously that this might not be something I'll add to the TeslaMatApi, since I prefer the application to stays a way of getting data from TeslaMate rather than feeding data into TeslaMate and it's database.

From a technical perspective that is doable to insert a new entry in the geofence table of TeslaMate with the various fields like long/lat, radius and pricing-methods. But I think that the use-case if fairly limited and more can go wrong than right.
Maybe it's better for you to make some postgresql-insert with some node-red library instead so that it fits your requirements.

Kind regards,
Tobias

**Comment by SteveDonie** [comment]
Thanks for the reply - I did end up using the postgresql node and just
doing a straight insert. Meets my needs.
